sql/slow-query.md

```diff
@@ -5,85 +5,208 @@ category: user guide
 
 ### 示例
 
-```
-2018/08/20 19:52:08.632 adapter.go:363: [warning] [SLOW_QUERY] cost_time:18.647928814s
-process_time:1m6.768s wait_time:12m11.212s backoff_time:600ms request_count:2058
-total_keys:1869712 processed_keys:1869710 succ:true con:3 user:root@127.0.0.1
-txn_start_ts:402329674704224261 database:test table_ids:[31],index_ids:[1],
-sql:select count(c) from sbtest1 use index (k_1)
+```sql
+# Time: 2019-03-18-12:10:19.513961 +0800
+# Txn_start_ts: 407078752230047745
+# User: root@127.0.0.1
+# Conn_ID: 1
+# Query_time: 16.479155653
+# Process_time: 5.634 Wait_time: 0.002 Request_count: 2 Total_keys: 20002 Process_keys: 20000
+# DB: test
+# Index_ids: [1]
+# Is_internal: false
+# Digest: 3635413fe0c8e1aa8307f4f018fe1a9325ea0b97452500106d3f6783fcb65e33
+# Num_cop_tasks: 10
+# Cop_proc_avg: 1 Cop_proc_p90: 2 Cop_proc_max: 3 Cop_proc_addr: 10.6.131.78
+# Cop_wait_avg: 5 Cop_wait_p90: 6 Cop_wait_max: 7 Cop_wait_addr: 10.6.131.79
+# Memory_max: 4096
+select * from t_slim, t_wide where t_slim.c0=t_wide.c0;
 ```
 
 ### 字段解析
 
-#### cost_time
-
-表示执行这个语句花费的时间。只有执行时间超过 slow-threshold 的语句才会输出这个日志。
-
-#### process_time
-
-表示这个语句在 TiKV 的处理时间之和，因为数据会并行的发到 TiKV 执行，这个值可能会超过 cost_time。
-
-#### wait_time
-
-表示这个语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队，当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
-
-#### backoff_time
-
-表示语句遇到需要重试的错误时在重试前等待的时间，常见的需要重试的错误有以下几种：遇到了 lock、Region 分裂、`tikv server is busy`。
-
-#### request_count
-
-表示这个语句发送的 Coprocessor 请求的数量。
-
-#### total_keys
-
-表示 Coprocessor 扫过的 key 的数量
-
-#### processed_keys
+* `Time`：表示日志打印时间。
+* `Txn_start_ts`：表示事务的开始时间戳，也是事务的 ID, 可以用这个值在日志中 grep 出事务相关的日志。
+* `User`：表示执行语句的用户名。
+* `Conn_ID`：表示 connection ID，即 session ID, 可以用类似 `con:3 ` 的关键字在  TiDB 日志中 grep 出 session ID 为 3 的日志。
+* `Query_time`：表示执行这个语句花费的时间。只有执行时间超过 slow-threshold 的语句才会输出这个日志，单位是秒，一下所有的时间字段的单位都是秒。
+* `Process_time`：执行 SQL  在 TiKV 的处理时间之和，因为数据会并行的发到 TiKV 执行，这个值可能会超过 `Query_time`。
+* `Wait_time`：表示这个语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队，当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
+* `Backoff_time`：表示语句遇到需要重试的错误时在重试前等待的时间，常见的需要重试的错误有以下几种：遇到了 lock、Region 分裂、`tikv server is busy`。
+* `Request_count`：表示这个语句发送的 Coprocessor 请求的数量。
+* `Total_keys`：表示 Coprocessor 扫过的 key 的数量。
+* `Process_keys`：表示 Coprocessor 处理的 key 的数量。相比 total_keys，processed_keys 不包含 MVCC 的旧版本。如果 processed_keys 和 total_keys 相差很大，说明旧版本比较多。
+* `DB`：表示当前的 database。
+* `Index_ids` ：表示语句涉及到的索引的 ID
+* `Is_internal`：表示是否是 TiDB 内部 SQL。true 为TiDB 内部执行的SQL, 比如 analyze，load variable 等；false 为用户执行的 SQL 。
+* `Digest`：表示 SQL 语句的指纹。
+* `Memory_max`：表示执行期间做多时候使用的内存数量, 单位为byte。
+* `Num_cop_tasks`：表示cop-tasks的数目。
+* `Cop_proc_avg`: cop-task的平均执行时间。
+* `Cop_proc_p90`: cop-task的P90分位执行时间。
+* `Cop_proc_max`: cop-task的最大执行时间。
+* `Cop_proc_addr`: 执行时间最长的cop-task所在地址。
+* `Cop_wait_avg`: cop-task的平均等待时间。
+* `Cop_wait_p90`: cop-task的P90分位等待时间。
+* `Cop_wait_max`: cop-task的最大等待时间。
+* `Cop_wait_addr`: 等待时间最长的cop-task所在地址。
```

Comment:
Please add necessary whitespaces between English words and Chinese characters for all sentences above